godel_simple_gui/src/blending_widget.cpp

```diff
@@ -0,0 +1,120 @@
+#include "godel_simple_gui/blending_widget.h"
+
+// UI specific
+#include "ui_blending_widget.h"
+// General ROS stuff
+#include <ros/console.h>
+#include "godel_msgs/SurfaceBlendingParameters.h"
+// States
+#include "godel_simple_gui/states/scan_teach_state.h"
+
+godel_simple_gui::BlendingWidget::BlendingWidget(QWidget* parent)
+  : QWidget(parent)
+  , active_state_(NULL)
+{
+  // UI setup
+  ui_ = new Ui::BlendingWidget;
+  ui_->setupUi(this);
+
+  options_ = new OptionsSubmenu();
+  options_->hide();
+
+  // Starts in scan teach state
+  changeState( new ScanTeachState() );
+
+  // Wire in buttons
+  connect(ui_->pushButtonNext, SIGNAL(clicked()), this, SLOT(onNextButton()));
+  connect(ui_->pushButtonBack, SIGNAL(clicked()), this, SLOT(onBackButton()));
+  connect(ui_->pushButtonReset, SIGNAL(clicked()), this, SLOT(onResetButton()));
+  connect(ui_->pushButtonOptions, SIGNAL(clicked()), this, SLOT(onOptionsButton()));
+
+  // Wire in Option signals
+  connect(options_, SIGNAL(saveRequested()), this, SLOT(onOptionsSave()));
+
+  // Connect to ROS services
+  loadParameters();
+}
+
+godel_simple_gui::BlendingWidget::~BlendingWidget()
+{
+  delete active_state_;
+  delete options_;
+}
+
+void godel_simple_gui::BlendingWidget::setText(const std::string& txt)
+{
+  ui_->textEditStatus->setPlainText(QString::fromStdString(txt));
+}
+
+void godel_simple_gui::BlendingWidget::onNextButton()
+{
+  active_state_->onNext(*this);
+}
+
+void godel_simple_gui::BlendingWidget::onBackButton()
+{
+  active_state_->onBack(*this);
+}
+
+void godel_simple_gui::BlendingWidget::onResetButton()
+{
+  active_state_->onReset(*this);
+}
+
+void godel_simple_gui::BlendingWidget::onOptionsButton()
+{
+  options_->show();
+}
+
+void godel_simple_gui::BlendingWidget::onOptionsSave()
+{
+}
```

Comment:
The save parameters in this simple gui are unimplemented. It's more of a "did not finish development" than an intentional omission.